packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit/Sources/in_app_purchase_storekit/StoreKit2/sk2_pigeon.g.swift

```diff
@@ -787,6 +789,26 @@ class InAppPurchase2APISetup {
     } else {
       isWinBackOfferEligibleChannel.setMessageHandler(nil)
     }
+    let isIntroductoryOfferEligibleChannel = FlutterBasicMessageChannel(
+      name:
+        "dev.flutter.pigeon.in_app_purchase_storekit.InAppPurchase2API.isIntroductoryOfferEligible\(channelSuffix)",
+      binaryMessenger: binaryMessenger, codec: codec)
+    if let api = api {
+      isIntroductoryOfferEligibleChannel.setMessageHandler { message, reply in
+        let args = message as! [Any?]
+        let productIdArg = args[0] as! String
+        api.isIntroductoryOfferEligible(productId: productIdArg) { result in
```

Comment:
Hi @LouiseHsu it looks like the extension where `isIntroductoryOfferEligible` is defined has this availability annotation:
https://github.com/flutter/packages/blob/80bf552a8de2fdc1e6c85fea2c0a0788261d05c5/packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit/Sources/in_app_purchase_storekit/StoreKit2/InAppPurchasePlugin%2BStoreKit2.swift#L7-L8)

Does this mean if the developer's app's target iOS version is < 15.0 they'll get a compile time error from the plugin code (even if they're not using these new APIs at all?)

Reply:
No, availability annotation on extensions will compile fine on lower device version, it just cannot reference any code inside that extension!